File: differ/src/main/java/ru/starksoft/differ/adapter/OnClickListener.java
package ru.starksoft.differ.adapter;

import android.os.Bundle;

import androidx.annotation.IntRange;
import androidx.annotation.NonNull;
import ru.starksoft.differ.viewmodel.ViewModel;

public interface OnClickListener {

	boolean onClick(@IntRange(from = 0) int position, @NonNull ViewModel viewModel, int action, @NonNull Bundle extra);

}

File: diff-adapter/src/main/java/ru/starksoft/differ/adapter/OnClickListener.kt
package ru.starksoft.differ.adapter

import android.os.Bundle
import androidx.annotation.IntRange
import ru.starksoft.differ.adapter.viewmodel.ViewModel

fun interface OnClickListener {

    fun onClick(@IntRange(from = 0) position: Int, viewModel: ViewModel, action: Int, extra: Bundle): Boolean
}